hs_core/views/__init__.py

```diff
@@ -779,8 +817,13 @@ def set_resource_flag(request, shortkey, *args, **kwargs):
 
     if request.META.get('HTTP_REFERER', None):
         request.session['resource-mode'] = request.POST.get('resource-mode', 'view')
+
+        web_services.update(shortkey)
+
         return HttpResponseRedirect(request.META.get('HTTP_REFERER', None))
 
+    web_services.update(shortkey)
```

Comment:
Need to put web_services.update() call into a celery task so it runs in a background thread so as not to block the main thread, especially given the calling functions are called very frequently.